src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -30,6 +31,8 @@ internal class SslStreamInternal
         private AsyncProtocolRequest _readProtocolRequest; // cached, reusable AsyncProtocolRequest used for read operations
         private AsyncProtocolRequest _writeProtocolRequest; // cached, reusable AsyncProtocolRequest used for write operations
 
+        private SemaphoreSlim _asyncWriteActiveSemaphore;
```

Comment:
Why is this needed?  The existing BeginWrite doesn't serialize on such a semaphore, does it?